src/git/Repository.ts

```diff
@@ -33,14 +34,22 @@ export class Repository {
         this.repoName = path.basename(path.resolve(repoPath));
     }
 
-    public static clone(toPath: string, remoteUrl: string, branch: string, depth: number): Promise<Repository> {
+    public static clone(toPath: string, repoProperties: IRepoStatus, depth: number): Promise<Repository> {
         return new Promise<Repository>((resolve, reject) => {
-            Global.isVerbose() && console.log('cloning branch', branch, 'from', remoteUrl, 'to', toPath);
-            const options = ['--branch', branch];
+            const options = [];
+
+            const refToCheckout = repoProperties.tag || repoProperties.latestTagForRelease || repoProperties.branch;
+            if (refToCheckout) {
+                Global.isVerbose() && console.log('cloning tag or branch', refToCheckout, 'from', repoProperties.url, 'to', toPath);
+                options.push('--branch', refToCheckout);
+            } else {
+                Global.isVerbose() && console.log('cloning default branch from', repoProperties.url, 'to', toPath);
+            }
             if (depth > 0) {
                 options.push('--depth', depth.toString(10));
             }
-            simpleGit().clone(remoteUrl, toPath, options, (err) => {
+            simpleGit().clone(repoProperties.url, toPath, options, (err) => {
+
```

Comment:
As discussed today: the only part missing is checking out the `release-version/1.2.3` branch name after cloning a tag/latestTagForRelease.